Fallback matcher must not match slashes

imod opened the pull request:
fixes https://github.com/microcks/microcks/issues/614

imod commented:
@lbroudoux I don't understand why the tests fail in this repo, but work on my fork for the exact same commit: https://github.com/imod/microcks/actions/runs/2476011578

lbroudoux commented:
Thanks for the PR! Seems like it fails on my machine too... Will try to investigate.

lbroudoux commented:
I think that failure is just due to your test being too deterministic ...
When no precise response is found, it defaults to first matching operation and mime-type. But 'first' is not precise in MongoDB terms. Depending on how the import was done and Mongo internals (or version, or arch), results may vary. The important thing is that you receive a response coming from the details operation with `details` in the description. I think the integration test should be simplified just to check that criterion.

lbroudoux commented on webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:
```diff
@@ -88,4 +88,46 @@ public void testSwaggerMocking() {
+         		+ "		\"street\":\"freestreet 3\","
+         		+ "		\"city\":\"Paris\""
+         		+ "}",
+               response.getBody(), JSONCompareMode.LENIENT);
```
I think this part should be simplified just to check that we have a response coming from the `details` operation. Something like `description.startsWith("Detail -")` should be enough.

lbroudoux commented on webapp/src/main/java/io/github/microcks/web/RestController.java:
```diff
@@ -143,7 +143,7 @@ public ResponseEntity<?> execute(
                   // Produce a matching regexp removing {part} and :part from pattern.
                   String operationPattern = getURIPattern(operation.getName());
-                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "(.)+");
+                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "([^/])+");
```
Shouldn't we have something similar on next line too?

imod replied:
yeah, thats probably true

lbroudoux reviewed (changes requested):
Need to make the test more robust by simplifying assertions.

imod commented:
hmm, not sure I follow...
I think there is only one example of `details` imported at all, therefore I also expect this one to be returned - but maybe I don't understand it completely

lbroudoux commented:
Actually I see 2 examples for the `details` responses in the OpenAPI file you attached to integration test. One for `Millefeuille` and another for `Eclair Cafe`...

imod commented:
@lbroudoux I simplified the test and adjusted the pattern for the second case too. The test are fine on my branch again, but fail on this one with an error which is related to a problem of the infrastructure. Maybe you can trigger the action again?
https://github.com/microcks/microcks/actions/runs/2487395986

imod commented:
@lbroudoux I resolved all the open points

lbroudoux commented on webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:
```diff
@@ -88,4 +91,32 @@ public void testSwaggerMocking() {
+      ResponseEntity<String> response = restTemplate.getForEntity("/rest/pastry-details/1.0.0/pastry/Millefeuille/details", String.class);
+      assertEquals(200, response.getStatusCode().value());
+      try {
+    	  JSONObject details = new JSONObject(response.getBody());
```
LGTM but code style has extra indent on 2nd and 3rd lines

lbroudoux commented on webapp/src/main/java/io/github/microcks/web/RestController.java:
```diff
@@ -143,8 +143,8 @@ public ResponseEntity<?> execute(
-                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "(.)+");
-                  operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "\\/(.+)");
+                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "([^/])+");
+                  operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "([^/])+");
```
I think this one is not OK as it removes first /.
`operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "\\/([^/]+)");` is the way to go. COuld you please modify this one ?

lbroudoux reviewed (changes requested):
Need a fix on postman notation and remove extra indents and we're good to merge!

imod commented on webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:
```diff
@@ -88,4 +91,32 @@ public void testSwaggerMocking() {
+         String description = details.getString("description");
+         assertTrue(description.startsWith("Detail -"));   
+      } catch (Exception e) {
+    	 fail("No Exception should be thrown here");
```
I don't know whats happening with these indents - copy paste just does not work on my site
Do you have a code formatter rule or anything I can use to apply?

imod commented:
sorry, this took a lot more commits then it should - some how the formatting was not displayed the same way on GH as in my IDE...

lbroudoux reviewed (approved):
Thanks!
